### ISSUE TITLE
Built from source nvim crashes on startup (MacOS)

### ISSUE THREAD

**Issue by VVKot** [neovim/neovim#15120]
Neovim started crashing for me when building from source starting from 3de777d259c379e9fffdf4ebc6aba4059b7400fc. `--HEAD` version from `brew` (v0.6.0-dev+48-g5377b2b00) is working fine.

- `nvim --version`: 5377b2b00aea1a0bde1b81452e6198dabe5b9796
- Operating system/version: MacOS BigSur 11.4
- Terminal name/version: iTerm2 3.4.8
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
sudo make clean && sudo make distclean && sudo make install
nvim -u NORC
```

### Actual behaviour
Neovim does crash. See crash report below.

```
Application Specific Information:
dyld2 mode
abort() called

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff204ce92e __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff204fd5bd pthread_kill + 263
2   libsystem_c.dylib             	0x00007fff20452411 abort + 120
3   libuv.1.dylib                 	0x000000010c1e2afa uv__signal_block_and_lock + 61
4   libuv.1.dylib                 	0x000000010c1e2568 uv__signal_start + 90
5   libuv.1.dylib                 	0x000000010c1e1439 uv_spawn + 473
6   libluv.1.dylib                	0x000000010c1ac117 luv_spawn + 2135
7   nvim                          	0x000000010beea93d lj_BC_FUNCC + 68

Thread 1:
0   libsystem_kernel.dylib        	0x00007fff204ccc4a kevent + 10
1   nvim                          	0x000000010beaf032 uv__io_poll + 1026 (kqueue.c:214)
2   nvim                          	0x000000010be9df91 uv_run + 433 (core.c:375)
3   nvim                          	0x000000010bc0127d loop_poll_events + 157 (loop.c:62)
4   nvim                          	0x000000010be34a7d tui_main + 1149 (tui.c:456)
5   nvim                          	0x000000010be48b10 ui_thread_run + 48 (ui_bridge.c:105)
6   libsystem_pthread.dylib       	0x00007fff204fd8fc _pthread_start + 224
7   libsystem_pthread.dylib       	0x00007fff204f9443 thread_start + 15

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x000000011210ee00  rcx: 0x00007ffee4102e58  rdx: 0x0000000000000000
  rdi: 0x0000000000000103  rsi: 0x0000000000000006  rbp: 0x00007ffee4102e80  rsp: 0x00007ffee4102e58
   r8: 0x0000000000000000   r9: 0x0000000000000008  r10: 0x000000011210ee00  r11: 0x0000000000000246
  r12: 0x0000000000000103  r13: 0x0000000000000003  r14: 0x0000000000000006  r15: 0x0000000000000016
  rip: 0x00007fff204ce92e  rfl: 0x0000000000000246  cr2: 0x000000010c1e595a
  
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133

Thread 0 instruction stream not available.

Thread 0 last branch register state not available.


Binary Images:
       0x10bafb000 -        0x10bfe6fff +nvim (0) <A08367D5-F849-3DC2-B875-C14E145690B7> /usr/local/bin/nvim
       0x10c196000 -        0x10c19ffff +libintl.8.dylib (0) <FA921CC0-395B-3155-8259-EA61DE25C5D2> /usr/local/opt/gettext/lib/libintl.8.dylib
       0x10c1a6000 -        0x10c1bdfff +libluv.1.dylib (0) <A3C71ED9-FA96-3E0B-B3FF-4CF08DBA2049> /usr/local/opt/luv/lib/libluv.1.dylib
       0x10c1d4000 -        0x10c1effff +libuv.1.dylib (0) <D5160993-D08E-376C-A444-680108D4BC1E> /usr/local/opt/libuv/lib/libuv.1.dylib
       0x112037000 -        0x1120d2fff  dyld (852) <1AC76561-4F9A-34B1-BA7C-4516CACEAED7> /usr/lib/dyld
    0x7fff20235000 -     0x7fff20236fff  libsystem_blocks.dylib (79) <48AF56A9-6E42-3A5E-A213-E6AFD8F81044> /usr/lib/system/libsystem_blocks.dylib
    0x7fff20237000 -     0x7fff2026cfff  libxpc.dylib (2038.120.1) <5751A7F5-6DC5-3090-B7F1-D90ED71BEF1F> /usr/lib/system/libxpc.dylib
    0x7fff2026d000 -     0x7fff20284fff  libsystem_trace.dylib (1277.120.1) <8E243C00-BFC2-3FAA-989C-0D72314DB04D> /usr/lib/system/libsystem_trace.dylib
    0x7fff20285000 -     0x7fff20322fff  libcorecrypto.dylib (1000.120.2) <FADB19A0-1BF3-3F47-B729-87B4FA8CA677> /usr/lib/system/libcorecrypto.dylib
    0x7fff20323000 -     0x7fff2034ffff  libsystem_malloc.dylib (317.121.1) <CAD162A5-7367-3A30-9C15-5D036411AEDE> /usr/lib/system/libsystem_malloc.dylib
    0x7fff20350000 -     0x7fff20394fff  libdispatch.dylib (1271.120.2) <7B229797-1F2E-3409-9D0C-060C7EEF2E12> /usr/lib/system/libdispatch.dylib
    0x7fff20395000 -     0x7fff203cefff  libobjc.A.dylib (824) <FE5AF22E-80A1-34BB-98D6-610879988BAA> /usr/lib/libobjc.A.dylib
    0x7fff203cf000 -     0x7fff203d1fff  libsystem_featureflags.dylib (28.60.1) <77F7F479-39BD-3111-BE3C-C74567FD120C> /usr/lib/system/libsystem_featureflags.dylib
    0x7fff203d2000 -     0x7fff2045afff  libsystem_c.dylib (1439.100.3) <38F8A126-C995-349A-B909-FF831914ED2E> /usr/lib/system/libsystem_c.dylib
    0x7fff2045b000 -     0x7fff204b0fff  libc++.1.dylib (905.6) <B3812B86-4FCF-3A10-8866-DF67940A974C> /usr/lib/libc++.1.dylib
    0x7fff204b1000 -     0x7fff204c6fff  libc++abi.dylib (905.6) <A0FE88B7-E157-3C9C-A29A-11D3BE3436B3> /usr/lib/libc++abi.dylib
    0x7fff204c7000 -     0x7fff204f6fff  libsystem_kernel.dylib (7195.121.3) <A4938CF5-ABC0-397B-8A6E-B7BEEFA24D0A> /usr/lib/system/libsystem_kernel.dylib
    0x7fff204f7000 -     0x7fff20502fff  libsystem_pthread.dylib (454.120.2) <17482C9D-061E-3769-AC9E-BE1239D33098> /usr/lib/system/libsystem_pthread.dylib
    0x7fff20503000 -     0x7fff2053efff  libdyld.dylib (852) <C10CEA28-D5A0-324F-8F07-8C7CE4805412> /usr/lib/system/libdyld.dylib
    0x7fff2053f000 -     0x7fff20548fff  libsystem_platform.dylib (254.80.2) <8664A4CD-EE27-3C71-B5CC-06E2B1B4F394> /usr/lib/system/libsystem_platform.dylib
    0x7fff20549000 -     0x7fff20574fff  libsystem_info.dylib (542.40.3) <EA3F9C9C-3116-3DB4-A3F1-5B03172C1E72> /usr/lib/system/libsystem_info.dylib
    0x7fff20575000 -     0x7fff20a12fff  com.apple.CoreFoundation (6.9 - 1776.103) <01EFB7F8-BCE6-32DF-A0B8-02F9027F882C> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff226de000 -     0x7fff2293dfff  libicucore.A.dylib (66112) <9957A773-012E-3ABA-9587-CFF787170AE8> /usr/lib/libicucore.A.dylib
    0x7fff2293e000 -     0x7fff22947fff  libsystem_darwin.dylib (1439.100.3) <BF5B5FD8-B5A3-3035-8641-466E625A6CE8> /usr/lib/system/libsystem_darwin.dylib
    0x7fff22d5d000 -     0x7fff22d68fff  libsystem_notify.dylib (279.40.4) <7FFECC25-FA84-3B59-9CC8-4D9DC84E6EC1> /usr/lib/system/libsystem_notify.dylib
    0x7fff24cf2000 -     0x7fff24d00fff  libsystem_networkextension.dylib (1295.120.5) <02486B74-EAAD-3055-AE20-F12E79B39297> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff24d5f000 -     0x7fff24d75fff  libsystem_asl.dylib (385) <B3E89650-A7FE-3E93-8A1B-D88145FDD45C> /usr/lib/system/libsystem_asl.dylib
    0x7fff2643a000 -     0x7fff26441fff  libsystem_symptoms.dylib (1431.120.1) <3BEA5355-D267-39D4-8BC6-A1703845BD3F> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff28448000 -     0x7fff28458fff  libsystem_containermanager.dylib (318.100.4) <2BBFF58C-D27E-3371-968D-7DE1E53749F6> /usr/lib/system/libsystem_containermanager.dylib
    0x7fff29160000 -     0x7fff29163fff  libsystem_configuration.dylib (1109.120.1) <C7A9BD10-192B-31D3-92ED-2581A61A99F6> /usr/lib/system/libsystem_configuration.dylib
    0x7fff29164000 -     0x7fff29168fff  libsystem_sandbox.dylib (1441.120.5) <DC075A7C-9D4A-32D3-9022-CD47764AFDAD> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff29e5a000 -     0x7fff29e5cfff  libquarantine.dylib (119.40.2) <4611645F-5817-3A80-8382-2DB03A8C0141> /usr/lib/system/libquarantine.dylib
    0x7fff2a405000 -     0x7fff2a409fff  libsystem_coreservices.dylib (127.1) <1E2DA16B-D528-3D43-86C2-2BB9127954A0> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff2a619000 -     0x7fff2a660fff  libsystem_m.dylib (3186.100.3) <21949128-D4E6-3179-B248-41B05C1CE102> /usr/lib/system/libsystem_m.dylib
    0x7fff2a661000 -     0x7fff2a661fff  libcharset.1.dylib (59) <4B3453D8-277A-38D3-862D-28DF71F3E285> /usr/lib/libcharset.1.dylib
    0x7fff2a662000 -     0x7fff2a667fff  libmacho.dylib (980) <3677B3B7-03E8-3804-B2FE-5640B18FE40E> /usr/lib/system/libmacho.dylib
    0x7fff2a684000 -     0x7fff2a68ffff  libcommonCrypto.dylib (60178.120.3) <BBA72D86-B9C1-3123-AE59-D629DE278695> /usr/lib/system/libcommonCrypto.dylib
    0x7fff2a690000 -     0x7fff2a69afff  libunwind.dylib (201) <3149D79A-911B-39ED-9C93-6C7E6B0860C7> /usr/lib/system/libunwind.dylib
    0x7fff2a69b000 -     0x7fff2a6a2fff  liboah.dylib (203.46) <0A17EAFC-15E9-37FE-8EE2-DE0F7F220AD8> /usr/lib/liboah.dylib
    0x7fff2a6a3000 -     0x7fff2a6adfff  libcopyfile.dylib (173.40.2) <7304CA0D-E93C-367F-9BEE-AC56B873F06C> /usr/lib/system/libcopyfile.dylib
    0x7fff2a6ae000 -     0x7fff2a6b5fff  libcompiler_rt.dylib (102.2) <0DB1902E-C79C-3E26-BE51-F70960ECF0B9> /usr/lib/system/libcompiler_rt.dylib
    0x7fff2a6b6000 -     0x7fff2a6b8fff  libsystem_collections.dylib (1439.100.3) <E180C04A-9CFB-3C8E-9C2B-978D23A99F2A> /usr/lib/system/libsystem_collections.dylib
    0x7fff2a6b9000 -     0x7fff2a6bbfff  libsystem_secinit.dylib (87.60.1) <8C33D323-C11C-34CB-9295-4D7C98B8AFD6> /usr/lib/system/libsystem_secinit.dylib
    0x7fff2a6bc000 -     0x7fff2a6befff  libremovefile.dylib (49.120.1) <6DEAEEC9-2A65-3C7B-A9CE-23245772FD07> /usr/lib/system/libremovefile.dylib
    0x7fff2a6bf000 -     0x7fff2a6bffff  libkeymgr.dylib (31) <FD167835-3829-3FFD-B13E-D18113E271AB> /usr/lib/system/libkeymgr.dylib
    0x7fff2a6c0000 -     0x7fff2a6c7fff  libsystem_dnssd.dylib (1310.120.71) <7BB607FE-EF79-3144-8BD0-A66792FF1443> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff2a6c8000 -     0x7fff2a6cdfff  libcache.dylib (83) <8B201058-2C34-3C12-9A7A-898CB0AAD150> /usr/lib/system/libcache.dylib
    0x7fff2a6ce000 -     0x7fff2a6cffff  libSystem.B.dylib (1292.120.1) <A8309074-31CC-31F0-A143-81DF019F7A86> /usr/lib/libSystem.B.dylib
    0x7fff2a6d0000 -     0x7fff2a6d3fff  libfakelink.dylib (3) <CF7D19AF-D162-369D-9501-0BEAC4D1188E> /usr/lib/libfakelink.dylib
    0x7fff2a6d4000 -     0x7fff2a6d4fff  com.apple.SoftLinking (1.0 - 1) <6C04D3E0-BFE0-32E2-A098-46D726F9B429> /System/Library/PrivateFrameworks/SoftLinking.framework/Versions/A/SoftLinking
    0x7fff2a70d000 -     0x7fff2a7fdfff  libiconv.2.dylib (59) <B9FD3BC7-6001-3E60-A7FB-CE8AAE07C805> /usr/lib/libiconv.2.dylib
    0x7fff2ae7e000 -     0x7fff2ae81fff  libutil.dylib (58.40.2) <4D8FD41B-89A5-31DA-BB0E-7F13C3B1652F> /usr/lib/libutil.dylib
    0x7fff2dad5000 -     0x7fff2dad5fff  liblaunch.dylib (2038.120.1) <FB6430FC-AACB-3AFF-8763-4C5AFABEF40E> /usr/lib/system/liblaunch.dylib
    0x7fff2ff64000 -     0x7fff2ff64fff  libsystem_product_info_filter.dylib (8.40.1) <D5194AB1-61C4-3C8D-9E3C-C65702BAB859> /usr/lib/system/libsystem_product_info_filter.dylib

```

### Expected behaviour
Neovim does not crash.

**Comment by VVKot** [comment]
Seems it is 3de777d259c379e9fffdf4ebc6aba4059b7400fc. cc @clason

**Comment by clason** [comment]
In fact, it's https://github.com/neovim/neovim/issues/15054

TL;DR: `brew uninstall luv` "fixes" the issue (until a CMake experts figures out _why_ this happens).
